Python/Module5_OddsAndEnds/Writing_Good_Code.ipynb

```diff
@@ -0,0 +1,1028 @@
+    "\n",
+    "It does not take long to experience the benefits of type-hinting through your IDE. This both accelerates your coding by informing you of the object types that you are working with on the fly, and helps to expose oversights in your code as soon as they are made. \n",
+    "\n",
+    "Finally, it is also worthwhile to highlight the [mypy](http://mypy-lang.org/) project, which is used to perform static type-checking on your code based on your type-hints. That is, mypy will automatically traverse your code and find potential bugs by identifying type conflicts in your code (e.g. trying to capitalize an integer) by checking their annotated and inferred types. This tool is most useful for large-scale code bases. Companies like Dropbox make keen use of mypy to identify inconsistencies in their code without having to hit runtime errors. Keep mypy in mind as you mature as a Python developer and find yourself working on projects of growing complexity."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "**Takeaway:**\n",
+    "\n",
+    "There is a simple syntax for annotating functions, class-methods, and variables with type-hints; this is a formal mechanism for documenting the types of objects that are expected as inputs to the function, as well as the return type of the function. It is critical to note that that type-hints are *never* enforced by Python - they only serve as a form of documentation.\n",
+    "\n",
+    "That being said, IDEs have powerful abilities to inspect type-hints and to highlight potential inconsistencies in your code. These capabilities can greatly facilitate your code-development. There are also third-party libraries like [mypy](http://mypy-lang.org/) that can be used to provide more rigorous type enforcement in your code.\n",
+    "\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using the `typing` Module\n",
+    "Although we can use Python's built-int type objects (e.g. `list` and `dict`) for type-hinting, the standard library's [typing](https://docs.python.org/3/library/typing.html) module provides objects that are used to form more descriptive hints. For example, suppose that we want to write a function that takes in a list of integers as an input. Using the built-in `list` object for our type-hint prevents us from indicating that the list should house integers. `typing.List`, on the otherhand, supports a richer syntax for type-hinting: `typing.List[int]` specifies \"a list of integers\". Familiarizing yourself with the `typing` module is the first step to writing good type hints.\n",
```

Comment:
Python's built-**in** type objects <-- not built in**t** type objects

`typing.List`, on the other hand, supports <-- space between other and hand. With this, I also suggest against starting the sentence with lowercase `typing`; maybe "Using `typing.List`, on the other hand" is better?

"...specifies "a list of integers." <-- period on the inside